main.go

```diff
@@ -1,54 +1,75 @@
 package e3dbClients
 
 import (
-    "context"
-    "encoding/json"
-    "fmt"
-    "net/http"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
 )
 
 // ClientConfig wraps configuration
 // needed by an e3db client
 type ClientConfig struct {
-    Host      string //Hostname of the e3db API to communicate with
-    APIKey    string //User/Client ID to use when communicating with the e3db API
-    APISecret string //User/Client secret to use when communicating with the e3db API
+	Host      string //Hostname of the e3db API to communicate with
+	APIKey    string //User/Client ID to use when communicating with the e3db API
+	APISecret string //User/Client secret to use when communicating with the e3db API
 }
 
 // E3DBHTTPAuthorizer implements the functionality needed to make
 // an authenticated call to an e3db endpoint.
 type E3DBHTTPAuthorizer interface {
-    AuthHTTPClient(ctx context.Context) *http.Client
+	AuthHTTPClient(ctx context.Context) *http.Client
 }
 
 // HTTPError wraps details of an HTTP error
 type HTTPError struct {
-    message    string
-    URL        string
-    StatusCode int
+	message    string
+	URL        string
+	StatusCode int
 }
 
 // Error implements the error interface for HTTPError.
 func (err *HTTPError) Error() string {
-    return err.message
+	return err.message
 }
 
 // MakeE3DBServiceCall attempts to call an e3db service by executing the provided request and deserializing the response into the provided result holder, returning error (if any).
 func MakeE3DBServiceCall(httpAuthorizer E3DBHTTPAuthorizer, ctx context.Context, request *http.Request, result interface{}) error {
-    client := httpAuthorizer.AuthHTTPClient(ctx)
-    response, err := client.Do(request)
-    if err != nil {
-        return err
-    }
-    defer response.Body.Close()
-    if !(response.StatusCode >= 200 && response.StatusCode <= 299) {
-        requestURL := request.URL.String()
-        return &HTTPError{
-            StatusCode: response.StatusCode,
-            URL:        requestURL,
-            message:    fmt.Sprintf("e3db: %s: server http error %d", requestURL, response.StatusCode),
-        }
-    }
-    err = json.NewDecoder(response.Body).Decode(&result)
-    return err
+	client := httpAuthorizer.AuthHTTPClient(ctx)
```

Comment:
So this and the next function look very nearly identical. Is there a reason not to combine the common code?

Reply:
They have distinctly different purposes so I like having them in different functions, but I could probably make a helper method to take out some of the repeated code as I just did with other methods in pdsClient. In this case I'm not sure if I see value, but maybe you see value in refactoring the two methods into one//adding a helper @galxy25?

Reply:
Definitely two methods as they will be used for distinctly  different purposes, but there would be value in extracting the common code into a helper.

Reply:
Sure -- helper seems like the right call here. The signatures of these functions are different, it just seems like this request pattern works exactly the same when it is used so may be worth abstracting. 👍